loader.mjs

```diff
@@ -63,6 +63,9 @@ const scalaJSHelpers = {
 
   // Closure
   closure: (f, data) => f.bind(void 0, data),
+  closureThis: (f, data) => function(...args) { return f(this, data, ...args); },
+  closureRest: (f, data, n) => ((...args) => f(data, ...args.slice(0, n), args.slice(n))),
+  closureThisRest: (f, data, n) => function(...args) { return f(this, data, ...args.slice(0, n), args.slice(n)); },
```

Comment:
It seems that the imported function for `closureRest` and `closureThisRest` receive  two args (`f` and `data`) but doesn't have third argument (`n`). Where this arg comes from?

```lisp
(import "__scalaJSHelpers" "closureRest" (func $__scalaJSHelpers#closureRest___fun (param (ref func)) (param anyref) (result (ref any))))
(import "__scalaJSHelpers" "closureThisRest" (func $__scalaJSHelpers#closureThisRest___fun (param (ref func)) (param anyref) (result (ref any))))
```

Also, how can we test these? it seems the test cases doesn't have a `restParameter`.

Reply:
Oops, good catch! I fixed it in the expression builder.

> Also, how can we test these? it seems the test cases doesn't have a `restParameter`.

As the comment in the tests mentions, currently we cannot test the `...rest` parameters. Although the IR does not require it, the source syntax that creates `Closure`s with rest params also requires Scala's full-blown `Seq`s (for vararags). We'll be able to add tests when we have enough things to support `Seq`.